src/registry/textarea/edit-validation.ts

```diff
@@ -0,0 +1,29 @@
+import {IntlShape} from 'react-intl';
+import {z} from 'zod';
+
+import {buildCommonSchema} from '@/registry/validation';
+
+const textareaSchema = z.string().optional();
+
+// case for when component.multiple=false
+const singleValueSchema = z
+  .object({multiple: z.literal(false)})
+  .and(z.object({defaultValue: textareaSchema}));
+
+// case for when component.multiple=true
+const multipleValueSchema = z
+  .object({multiple: z.literal(true)})
+  .and(z.object({defaultValue: textareaSchema.array()}));
+
+const textareaSpecific = z.object({
+  validate: z.object({
+    maxLength: z.number(),
+  }),
+});
```

Comment:
Am I missing anything? Should I also add textarea specific fields?

Reply:
many components don't necessarily have specific validations, I think this one is even overkill because the max length validation input should only allow numbers in the first place.

*if* we go the validation route here (which adds extra robustness), it should be as specific as possible:

```tsx
z.number().int().gte(1).optional()
```
to ensure that we don't get floats, a value is at least 1 if provided and the `.optional()` to allow it to be unspecified.